examples/waypoint_tracking_agent.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import random
+
+import gym
+
+from smarts.core.agent import Agent, AgentSpec
+from smarts.core.agent_interface import AgentInterface, AgentType, DoneCriteria
+from smarts.core.sensors import Observation
+from smarts.core.utils.episodes import episodes
+from smarts.env.wrappers.single_agent import SingleAgent
+
+# The following ugliness was made necessary because the `aiohttp` #
+# dependency has an "examples" module too.  (See PR #1120.)
+if __name__ == "__main__":
+    from argument_parser import default_argument_parser
+else:
+    from .argument_parser import default_argument_parser
+
+logging.basicConfig(level=logging.INFO)
+
+AGENT_ID = "WaypointTrackingAgent"
+
+
+class WaypointTrackingAgent(Agent):
+    def __init__(self):
+        self.waypoint_path = []
+
+    def act(self, obs):
+
+        current_lane = obs.ego_vehicle_state.lane_index
+        # Desired speed is in m/s
+        desired_speed = 10
+        wp_index = 0
+
+        num_lanes = 0
+        for path in obs.waypoint_paths:
+            num_lanes = max(num_lanes, path[0].lane_index + 1)
+
+        # This is when there are waypoint paths that change lanes
+        if (
+            len(obs.waypoint_paths) > num_lanes
+            and len(obs.waypoint_paths) <= num_lanes ** 2
```

Comment:
Can we say this can accurately find whether we have a waypoint path that has a lane change?

Reply:
I am not sure, my reasoning was that by default there must be an equal number of waypoint paths as lanes, and then any lane changing path would be in addition to the number of lanes. And then the maximum would be the number of lanes squared unless there are multiple waypoint paths that start and end in the same lane.